app/javascript/components/dialog-editor/TagControl.jsx

```diff
@@ -0,0 +1,51 @@
+import React, { useState, useEffect } from 'react';
+import { componentTypes } from '@data-driven-forms/react-form-renderer';
+import { formFieldsMapper } from '@data-driven-forms/pf3-component-mapper';
+
+import { API } from '../../http_api';
+
+const Component = formFieldsMapper[componentTypes.SELECT];
+const url = id => `/api/categories/${id}`;
+
+export const RawTagControl = ({ categoryId, forceSingleValue, ...props }) => {
+  const [{ multi, loadOptions }, setState] = useState({
+    multi: true,
+    loadOptions: () => new Promise(() => undefined),
+  });
+
+  useEffect(() => {
+    let isSubscribed = true;
+
+    if (categoryId) {
+      API.get(`${url(categoryId)}?attributes=single_value`).then(({ single_value: singleValue }) => {
+        if (isSubscribed) {
+          setState({
+            multi: !(singleValue || forceSingleValue),
+            loadOptions: () =>
+              API.get(
+                `${url(categoryId)}/tags?expand=resources&attributes=categorization`,
+              ).then(({ resources }) =>
+                resources.map(
+                  ({ categorization: { id: value, description: label } }) => ({
+                    label,
+                    value,
+                  }),
+                )),
+          });
+        }
+      });
+    }
+
+    return () => { isSubscribed = false; };
+  }, [categoryId]);
+
+  return <Component multi={multi} loadOptions={loadOptions} {...props} />;
+};
+
+const TagControl = ({ FieldProvider, ...rest }) => (
+  <FieldProvider {...rest}>
+    {({ ...props }) => <RawTagControl {...props} /> }
```

Comment:
no need for copy in argument

Reply:
I don't understand this, please explain.

Reply:
I don't understand what you're saying :confused:

Reply:
I don't understand what you're saying confused

Reply:
```JSX
 { props => <RawTagControl {...props} /> }
```

Reply:
Thanks :+1: